### PR TITLE
Single node cluster provider / lookup

### PR DESCRIPTION
**Pull request by mhelleborg**
## Description

Added single node 'cluster' provider, allowing Proto.Cluster to be used in single applications with the cluster / virtual actor abstractions. 

This makes it easy to get started, and allows for the same programming model in a single service use case as in a clustered environment. It will not work with multiple members, but is meant to simplify local testing, and use cases where you only ever have a single instance of the application.

## Purpose
This pull request is a:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    private readonly Cluster _cluster;
+    private readonly Dictionary<ClusterIdentity, PID> _actors = new();
+    private readonly HashSet<ClusterIdentity> _inFlightIdentityChecks = new();
+
+    public SingleNodeActivatorActor(Cluster cluster) => _cluster = cluster;
+
+    private Task OnStarted(IContext context)
+    {
+        var self = context.Self;
+        _cluster.System.EventStream.Subscribe<ActivationTerminated>(e => _cluster.System.Root.Send(self, e));
```
Comment (marcinbudny) [inline src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs]: Could this be simplified by using the `Subscribe<TMsg>(ISenderContext context, params PID[] pids)` overload?

File: src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+        if (!_actors.ContainsKey(msg.ClusterIdentity))
+            return Task.CompletedTask;
+
+        if (Logger.IsEnabled(LogLevel.Trace))
+            Logger.LogTrace("[SingleNode] Terminating {Pid}", msg.Pid);
+
+        _actors.Remove(msg.ClusterIdentity);
+
+        // Broadcast ActivationTerminated to all nodes so that PidCaches gets
```
Comment (marcinbudny) [inline src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs:93]: Perhaps change the comment to reflect the fact that there is a single node? Also just send the event to the EventStream directly, skipping the broadcast (to make the code easier to reason about).

File: src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs (hunk 1 above)
Comment (mhelleborg) [reply src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs]: Yes, fixed :)

File: src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs (hunk 2 above)
Comment (mhelleborg) [reply src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs:93]: Cleared up, thanks 😄

**Pull request merged**